api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -439,6 +441,28 @@ func validateLoadBalancerProfile(loadBalancerProfile *LoadBalancerProfile, fldPa
 	return allErrs
 }
 
+func validateAMCPVirtualNetwork(amcpSpec AzureManagedControlPlaneSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// amcpSpec.VirtualNetwork.ResourceGroup is not set, it means the virtual network is in the same resource group as the control plane.
+	if amcpSpec.ResourceGroupName != amcpSpec.VirtualNetwork.ResourceGroup {
+		amcpVirtualNetwork := amcpSpec.VirtualNetwork
+		// CIDR blocks get defaulted in the defaulting webhook, so we can assume they are always set.
+		_, parentNet, err := net.ParseCIDR(amcpVirtualNetwork.CIDRBlock)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("CIDRBlock"), amcpVirtualNetwork.CIDRBlock, "pre-existing virtual networks CIDR block is wrong"))
+		}
+		subnetIP, _, err := net.ParseCIDR(amcpVirtualNetwork.Subnet.CIDRBlock)
```

Comment:
For my own understanding, why are we not using some of the returned values on this line as well as line 453?

Reply:
From https://pkg.go.dev/net#ParseCIDR
`net.ParseCIDR(...)` returns IP address and the network implied by the IP and prefix length. For example, ParseCIDR("192.0.2.1/24") returns the IP address 192.0.2.1 and the network 192.0.2.0/24.
So, in-addition to checking if the subnet CIDR is valid, we are checking if the starting IP is part of VNet CIDR.

Reply:
This really is a soft check. I wonder if we should enforce by checking ending IP as well.

Reply:
I don't think ending IP check if required since a pre-existing VNet's creation enforces that the subnet created, as part of that VNet, has a subnet CIDR within VNet's CIDR. 
An ending UP check will be an overkill for a webhook.